mmv1/products/spanner/BackupSchedule.yaml

```diff
@@ -0,0 +1,147 @@
+  guides:
+    'Official Documentation': 'https://cloud.google.com/spanner/docs/backup'
+  api: 'https://cloud.google.com/spanner/docs/reference/rest/v1/projects.instances.databases.backupSchedules'
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  encoder: templates/terraform/encoders/spanner_schedule_backup.go.erb
+  decoder: templates/terraform/decoders/spanner_schedule_backup.go.erb
+  pre_update: templates/terraform/pre_update/spanner_schedule_backup.go.erb
+id_format: 'projects/{{project}}/instances/{{instance}}/databases/{{database}}/backupSchedules/{{name}}'
+import_format:
+  - 'projects/{{project}}/instances/{{instance}}/databases/{{database}}/backupSchedules/{{name}}'
+  - '{{project}}/{{instance}}/{{database}}/{{name}}'
+  - '{{instance}}/{{database}}/{{name}}'
+docs: !ruby/object:Provider::Terraform::Docs
+  warning: |
+    This resource creates a Spanner Backup Schedule on a project that already has
+    a Spanner database.
+    This resource is owned by the database it is backing up, and is deleted along
+    with the database. The actual backups are not though.
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'spanner_backup_schedule_daily_full'
+    primary_resource_id: 'full-backup'
+    vars:
+      database_name: 'database-id'
+      deletion_protection: 'true'
+    test_vars_overrides:
+      deletion_protection: 'false'
+    oics_vars_overrides:
+      deletion_protection: 'false'
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'spanner_backup_schedule_daily_incremental'
+    primary_resource_id: 'incremental-backup'
+    vars:
+      database_name: 'database-id'
+      deletion_protection: 'true'
+    test_vars_overrides:
+      deletion_protection: 'false'
+    oics_vars_overrides:
+      deletion_protection: 'false'
+parameters:
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'instance'
+    resource: 'Instance'
+    imports: 'name'
+    description: 'The instance to create the database on.'
+    immutable: true
+    required: true
+    url_param_only: true
+    custom_expand: 'templates/terraform/custom_expand/resourceref_with_validation.go.erb'
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'database'
+    resource: 'Database'
+    imports: 'name'
+    description: 'The database to create the backup schedule on.'
+    immutable: true
+    required: true
+    url_param_only: true
+    custom_expand: 'templates/terraform/custom_expand/resourceref_with_validation.go.erb'
+properties:
+  - !ruby/object:Api::Type::String
+    name: name
+    output: true
+    description: |
+      The unique backup schedule identifier across all locations and databases for the given project, which cannot be changed after
+      the schedule is created. Values are of the form [a-z][-a-z0-9]*[a-z0-9].
+    immutable: true
+    custom_flatten: templates/terraform/custom_flatten/name_from_self_link.erb
+  - !ruby/object:Api::Type::String
+    name: retentionDuration
```

Comment:
For these mutable fields, we would normally like to see update tests to prove they can be changed: https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-an-update-test. Is there a reason that can't be done for this resource?

Reply:
Added tests